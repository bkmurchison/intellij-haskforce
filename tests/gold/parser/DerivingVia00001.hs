module DerivingVia00001 where

newtype Foo = Foo Bar
  deriving stock (Generic, Eq, Show)
  deriving anyclass (Spam)
  deriving (ToJSON, FromJSON) via Bar

deriving via Eggs instance ToJSON Baz
