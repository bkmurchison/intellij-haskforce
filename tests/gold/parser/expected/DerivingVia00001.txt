Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('DerivingVia00001')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    NEWTYPEDECL
      PsiElement(HaskellTokenType.newtype)('newtype')
      TYCON
        CONID
          PsiElement(HaskellTokenType.conidRegexp)('Foo')
      PsiElement(HaskellTokenType.=)('=')
      NEWCONSTR
        CON
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('Foo')
        ATYPE
          OQTYCON
            QTYCON
              TYCON
                CONID
                  PsiElement(HaskellTokenType.conidRegexp)('Bar')
      PsiElement(HaskellTokenType.deriving)('deriving')
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('stock')
      PsiElement(HaskellTokenType.()('(')
      QTYCLS
        TYCLS
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('Generic')
      PsiElement(HaskellTokenType.,)(',')
      QTYCLS
        TYCLS
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('Eq')
      PsiElement(HaskellTokenType.,)(',')
      QTYCLS
        TYCLS
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('Show')
      PsiElement(HaskellTokenType.))(')')
      PsiElement(HaskellTokenType.deriving)('deriving')
      VARID
        PsiElement(HaskellTokenType.varidRegexp)('anyclass')
      PsiElement(HaskellTokenType.()('(')
      QTYCLS
        TYCLS
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('Spam')
      PsiElement(HaskellTokenType.))(')')
      PsiElement(HaskellTokenType.deriving)('deriving')
      PsiElement(HaskellTokenType.()('(')
      QTYCLS
        TYCLS
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('ToJSON')
      PsiElement(HaskellTokenType.,)(',')
      QTYCLS
        TYCLS
          CONID
            PsiElement(HaskellTokenType.conidRegexp)('FromJSON')
      PsiElement(HaskellTokenType.))(')')
      PsiElement(HaskellTokenType.varidRegexp)('via')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Bar')
    DERIVINGDECL
      PsiElement(HaskellTokenType.deriving)('deriving')
      PsiElement(HaskellTokenType.varidRegexp)('via')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Eggs')
      PsiElement(HaskellTokenType.instance)('instance')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('ToJSON')
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Baz')